### PR TITLE
chore: useERC721TokenDetailedOwnerList

### PR DESCRIPTION
**Pull request by zhouhanseng**

### CODE REVIEW

**Comment by Jack-Works** [comment]
@zhouhanseng why you made so many lockfile changes inside this PR?

**Comment by zhouhanseng** [comment]
@Jack-Works reverted

File: packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+async function getERC721TokenDetailedOwnerListFromOpensea(
+    contractDetailed: ERC721ContractDetailed,
+    owner: string,
+    apiURL: string,
+    offset: number,
+) {
+    const response = await fetch(
+        `${apiURL}?owner=${owner}&asset_contract_address=${contractDetailed.address}&limit=50&offset=${offset * 50}`,
```
Comment (guanbinrui) [inline packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts:100]: Try to construct this complex URL with `URLSearchParams`.

Comment (zhouhanseng) [reply packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts:100]: okay

File: packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    apiURL: string,
+    offset: number,
+) {
+    const response = await fetch(
+        urlcat(apiURL, {
+            owner,
+            asset_contract_address: contractDetailed.address,
+            limit: '50',
+            offset: String(offset * 50),
+        }),
```
Comment (septs) [inline packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts:106]:
```suggestion
        urlcat(apiURL, {
            owner,
            asset_contract_address: contractDetailed.address,
            limit,
            offset: offset * limit,
        }),
```

plz extract result as variable, and add `limit` variable.

File: packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            (asset) =>
+                ({
+                    tokenId: asset.token_id,
+                    contractDetailed,
+                    info: {
+                        name: asset.name,
+                        image: asset.image_url,
+                        owner,
+                    },
+                } as ERC721TokenDetailed),
```
Comment (septs) [inline packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts:126]:
```suggestion
            (asset): ERC721TokenDetailed =>
                ({
                    tokenId: asset.token_id,
                    contractDetailed,
                    info: {
                        name: asset.name,
                        image: asset.image_url,
                        owner,
                    },
                }),
```

File: packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            limit,
+            offset: offset * limit,
+        }),
+    )
+    type DataType = {
+        image_url: string
+        name: string
+        token_id: string
+    }
+    if (response.ok) {
```
Comment (septs) [inline packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts:114]:
```suggestion
    if (!response.ok) {
       return
    }
```

File: packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+async function getERC721TokenDetailedOwnerListFromOpensea(
+    contractDetailed: ERC721ContractDetailed,
+    owner: string,
+    apiURL: string,
+    offset: number,
+) {
+    const limit = 50
+    const response = await fetch(
+        urlcat(apiURL, {
```
Comment (septs) [inline packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts:102]:
```ts
const detailedURL = urlcat(....)
fetch(detailedURL)
```

File: packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        const tokenId = await safeNonPayableTransactionCall(
+            erc721TokenContract.methods.tokenOfOwnerByIndex(owner, i + offset * queryLimit),
+        )
+
+        if (tokenId) {
+            const tokenDetailed = await getERC721TokenDetailedFromChain(contractDetailed, erc721TokenContract, tokenId)
+            return tokenDetailed
+        }
+
+        return
```
Comment (septs) [inline packages/web3-shared/src/hooks/useERC721TokenDetailedOwnerList.ts:84]:
```suggestion
        if (!tokenId) {
            return
        }

        return getERC721TokenDetailedFromChain(contractDetailed, erc721TokenContract, tokenId)
```

**Review by septs — APPROVED** [review]

**Review by NikoManiac — APPROVED** [review]

**Pull request merged**